### PR TITLE
Add Pod securityContext for target allocator

### PR DESCRIPTION
**Pull request by husnialhamdani**
**Description:** <Describe what has changed.>
Add Pod securityContext to target allocator as it was removed in previous changes.

**Link to tracking Issue:** <Issue number if applicable>
#2495 

**Testing:** <Describe what testing was performed and which tests were added.>
Added a unit and E2E test to specifically check fields unique to the Pod security context.

### CODE REVIEW

File: .chloggen/fix-ta-pod-security-context.yaml [hunk 1]
```diff
@@ -0,0 +1,16 @@
+# One of 'breaking', 'deprecation', 'new_component', 'enhancement', 'bug_fix'
+change_type: breaking
```
Comment (jaronoff97) [inline .chloggen/fix-ta-pod-security-context.yaml:2]: this is an enhancement as you are adding fields here.

File: tests/e2e/targetallocator-features/00-install.yaml [hunk 2]
```diff
@@ -51,6 +51,7 @@ spec:
     securityContext:
       runAsUser: 1000
       runAsGroup: 3000
+      fsGroup: 3000
```
Comment (jaronoff97) [inline tests/e2e/targetallocator-features/00-install.yaml:54]: i believe you need to set this in a different block for the podSecurityContext

Comment (husnialhamdani) [reply tests/e2e/targetallocator-features/00-install.yaml]: Is it good now?

File: tests/e2e/targetallocator-features/00-assert.yaml [hunk 3]
```diff
@@ -56,6 +56,8 @@ spec:
                     operator: In
                     values:
                       - "true"
+      securityContext:
+        runAsUser: 1000
```
Comment (jaronoff97) [inline tests/e2e/targetallocator-features/00-assert.yaml:60]: should we assert the other properties here too?
```
      runAsGroup: 3000
      fsGroup: 2000
```

Comment (husnialhamdani) [reply tests/e2e/targetallocator-features/00-assert.yaml:60]: added

**Review by jaronoff97 — APPROVED** [review]

**Pull request merged**